aiohttp/cookiejar.py

```diff
@@ -234,7 +239,6 @@ def filter_cookies(
         self, request_url: URL = URL()
     ) -> Union["BaseCookie[str]", "SimpleCookie[str]"]:
         """Returns this jar's cookies filtered by their attributes."""
-        self._do_expiration()
```

Comment:
I think this'll conflict with #7777, so I'd suggest leaving it in for now. Then that PR should end up removing the iter over self.

Reply:
This PR should be unblocked now.